src/managers/GroupManager.ts

```diff
@@ -82,6 +85,44 @@ export default class GroupManager extends BaseManager<Group> implements GroupMan
         return group
     }
 
+    /**
+     * Joins a group.
+     *
+     * @param inviteLinkOrGroupID The group invite link or group ID.
+     * @param shareToken The group's share token.
+     * @returns The joined group.
+     */
+    join(inviteLink: string): Promise<Group>
+    join(groupID: string, shareToken: string): Promise<Group>
+    public async join(inviteLinkOrGroupID: string, shareToken?: string): Promise<Group> {
+        if (shareToken !== undefined) {
+            return await this.joinWithToken(inviteLinkOrGroupID, shareToken)
+        } else {
+            const urlPath = new URL(inviteLinkOrGroupID).pathname
+            const matches = urlPath.match(/.+\/(\d+)\/([A-Za-z0-9]+)$/)
+            if (matches === null) throw new Error(`Invalid invite link\n-- URL: ${inviteLinkOrGroupID}`)
+            return await this.joinWithToken(matches[1], matches[2])
+        }
+    }
+
+    private async joinWithToken(groupID: string, shareToken: string): Promise<Group> {
+        const res = await this.client.rest.api<PostJoinGroupResponse>('POST', `groups/${groupID}/join/${shareToken}`)
+        const group = this._upsert(new Group(this.client, res.group))
+        if (res.group.members) {
+            res.group.members.forEach(data => {
+                const user = this.client.users._upsert(
+                    new User(this.client, {
+                        id: data.user_id,
+                        avatar: data.image_url,
+                        name: data.name,
+                    }),
+                )
+                group.members._upsert(new Member(this.client, group, user, data))
```

Comment:
this block of code gives me deja vu every time... we should (eventually) abstract this away into the `Group` constructor or add a utility method to either `Group` or `MemberManager` class to handle adding members to a group (and the user cache) from an API response. This can be handled in a separate issue though so no need to change anything here for now.